lightrag/lightrag/components/model_client/lite_client.py

```diff
@@ -0,0 +1,175 @@
+
+        validate = litellm.validate_environment(self.model)
+        if not validate["keys_in_environment"]:
+            ApiError = validate["missing_keys"]
+            raise ValueError(f"Environment variable {ApiError} must be set")
+        return litellm
+
+    def parse_chat_completion(self, response: Any) -> str:
+        """Parse completion output to strings"""
+        log.debug(f"completion: {response}")
+        if "choices" in response:
+            return response["choices"][0]["message"]["content"]
+        else:
+            log.error(f"Error parsing the completion: {response}")
+            raise ValueError(f"Error parsing the completion: {response}")
+
+    def parse_embedding_response(
+        self, response: Dict[str, List[float]]
+    ) -> EmbedderOutput:
+        """Parse the embedding response to a structure LightRAG components can understand."""
+        try:
+            embeddings = Embedding(embedding=response["data"][0]["embedding"], index=0)
+            return EmbedderOutput(data=[embeddings])
+        except Exception as e:
+            log.error(f"Error parsing the embedding response: {e}")
+            return EmbedderOutput(data=[], error=str(e), raw_response=response)
+
+    def convert_inputs_to_api_kwargs(
+        self,
+        input: Optional[Any] = None,
+        model_kwargs: Dict = {},
+        model_type: ModelType = ModelType.UNDEFINED,
+    ) -> Dict:
+        """Convert the Component's standard input and model_kwargs into API-specific format"""
+        final_model_kwargs = model_kwargs.copy()
+        supported_params = get_supported_openai_params(self.model)
+
+        if model_type == ModelType.EMBEDDER:
+            if isinstance(input, str):
+                input = [input]
+            if not isinstance(input, List):
+                raise TypeError("input must be a string or a list of strings")
+            final_model_kwargs["input"] = input
+        elif model_type == ModelType.LLM:
+            messages: List[Dict[str, str]] = []
+            if input is not None and input != "":
+                messages.append(
+                    {
+                        "role": "user",
+                        "content": input,
+                    }
+                )
+            final_model_kwargs["messages"] = messages
+        else:
+            raise ValueError(f"model_type {model_type} is not supported")
+
+        # Filter out unsupported params
+        if self.drop_params:
+            final_model_kwargs = {
+                k: v for k, v in final_model_kwargs.items() if k in supported_params
+            }
+
+        return final_model_kwargs
+
+    @backoff.on_exception(
+        backoff.expo,
+        (Timeout, APIConnectionError, InternalServerError, RateLimitError),
+        max_time=5,
+    )
+    def call(self, api_kwargs: Dict = {}, model_type: ModelType = ModelType.UNDEFINED):
+        if "model" not in api_kwargs:
+            api_kwargs["model"] = self.model

```

Comment:
This will not make it to the actual call if the model_kwargs are actually set, because init_sync/async_client doesn't actually check to see if the model_kwargs are set.

Reply:
ok